src/repr/src/row/encoding.rs

```diff
@@ -155,7 +154,6 @@ impl RowPacker<'_> {
                 Some(ProtoDatumOther::False) => self.push(Datum::False),
                 Some(ProtoDatumOther::True) => self.push(Datum::True),
                 Some(ProtoDatumOther::JsonNull) => self.push(Datum::JsonNull),
-                Some(ProtoDatumOther::Dummy) => self.push(Datum::Dummy),
```

Comment:
migration-wise, this is the critical line. if we did write these down before (you're much better positioned to know this than I am), then this will fall through to the `return Err(format!("unknown datum type: {}", o))` case below. persist_source seems to panic on that [1], I dunno if there are other callers and what they do

[1]: https://github.com/MaterializeInc/materialize/blob/47483b2c089f1e9470991b2ef271ff3465e88ab4/src/storage/src/source/persist_source.rs#L363-L366

> but it's probably some amount of breaking, and probably there is some reserved thing I have to do in any case

yeah, this is exactly right. you want an enum reserved value, at least until the next time we wipe all envs https://developers.google.com/protocol-buffers/docs/proto3#reserved_values